### ISSUE TITLE
shall we increase the size of weight now

### ISSUE THREAD

**Issue by l1t1** [leela-zero/leela-zero#2477]
it seems the grow of strength is getting harder at current size

**Comment by l1t1** [comment]
a paper https://www.nature.com/articles/s41467-019-11786-6

**Comment by Glrr** [comment]
Afaik, @gcp already talked about increasing playouts and reducing learning rate. After that, the next step would probably be to give way to a new generation of engines (eg katago).

**Comment by jillybob** [comment]
A good halfway is to train 30b from these 40b games and generate games at 2400 visits with 30b net

**Comment by Marcin1960** [comment]
> 
> 
> A good halfway is to train 30b from these 40b games and generate games at 2400 visits with 30b net

Excellent idea. For very, very long LZ was based on the assumption that bigger (and slower) is always better. Now it is time to see if it is not the case. 30b seems perfect to try!!!

**Comment by jillybob** [comment]
The best net on time parity is somewhere between 20b and 40b. 
[The best 20b ( ELF V2) is between LZ 226 and LZ 238 at time parity ](https://github.com/breakwa11/GoAIRatings)

[238 results here ](http://www.yss-aya.com/cgos/19x19/cross/LZ_05db_ELFv2_p800.html)

It has taken 12 months for 40b to catch up to a maximum trained 20b at time parity. Now what if we have 25b or 30b maximum trained using 40b games? It will very likely be stronger than LZ 40b at time parity. So we can generate training games faster, and speed boost 40b training, like how ELF training boosted 40b training.

**Comment by Ishinoshita** [comment]
Incidentally, the advantage of self-playing with a smaller network but at higher visits might be to help generating policy training targets with higher granularity, thus less threshold effect for policy due to quantitization (targets are multiples of 1/1600, at min) and dirichlet noise (the more the visits, the less the noise imprint on final distribution, apart from genuinely good discoveries).

*Though that might be not relevant at all when moving just from 1600v to 2400v.*

*Edit: Possibly, the more important effect might be better learning of tactical situations*

**Comment by lzytek** [comment]
Some random thoughts:
Roughly speaking the larger the size, the better coverage of the problem domain. Once you get a lot of samples and regress to smaller size, hopefully you can get better abstractions. IMHO, it's a back and forth process. Also, for Go specific, majority abstractions are local, so we can focus on training locals (corners vs. center) etc. Like modern H.265 video encoding, it's not evenly distributed.

**Comment by wpstmxhs** [comment]
Please be aware that almost all of this comment are my guesses.
I want to say two episodes first.

First
----

In Fox Go Server, there was a very strong computer go player named *baldeagle* before competition on August.
There are some guesses that *baldeagle* on *Fox Go Server* was *FineArt* because its unique move or playing style is same as *FineArt* of competition.

![1566820972](https://user-images.githubusercontent.com/28377280/63800640-8a347500-c949-11e9-882a-550697db9b75.png)
<p align="center">
A same move from baldeagle and FineArt. Other computer go engines don't play there.
</p>

And *baldeagle* lost several games from *KataGo*. *KataGo* is made to count territory more exactly than current Zero-style programs.

Second
----

In Korea, there are many online Go playing platforms. *Hangame Baduk* is one of them. *Hangame Baduk* allows members to play with *HanDol* at any time.

I tested *Octopus Go OX24* on *HanDol*. *OX24* playing white consistently beats *HanDol*. Although *HanDol* on there may be weaker than its competition version (due to its slower machine), *HanDol* is still strong enough to beat *MiniGo* or *Leela Zero*. *Leela Zero* is difficult to beat it consistently. pangafu/LeelaMasterWeight#18

I think that *Octopus Go OX24* is trained considerately about counting territory, but I'm not sure about that. (I do not know the details of Octopus Go's training procedure much.)

My suggestion
----
A lesson from two episodes may lead to a suggestion about changing architecture.
I think an additional architecture like score head in KataGo is necessary for stronger Leela Zero.
For complex situations, more output head may help.
How do you think?

**Comment by Ishinoshita** [comment]
Adding new architecture or training features from Katago, like score or playout cap oscillation, sounds indeed way more attractive than just putting more blocks, which probably wouldn't help much.

**Comment by alreadydone** [comment]
A Golaxy representative actually said with much assurance that *baldeagle* is FineArt and they knew that because they saw how FineArt played in the matches. http://www.newsmth.net/nForum/#!article/Weiqi/631362?p=2
BTW, LZ is able to find the shown move with millions of visits; in one instance, S16 had 2.3m visits (42.2%) when P9 had 5.0m visits (43.1%).

My best suggestion is to implement this method of increasing model capacity without affecting inference time: https://github.com/lightvector/KataGo/issues/42#issuecomment-513102126

**Comment by wpstmxhs** [comment]
@Ishinoshita 
I agree with you. Currently, Leela Zero has only two output heads, one is policy head and the other one is value head. but at the high level like current Leela Zero, it seems it is harder to train outputs well. for example, in the case of value head, it is harder to estimate the situation. As you know, in the game of Go, scoring territories is necessary for estimation of situation. because goal of Go is having more score[territory] than the opponent.

So I guess mechanisms about score (i.e. score head like KataGo or score-centric training algorithm like Octopus Go) may help to let weights know that getting more score leads to advantage in the game. Only using one value head to estimate the game situation has a potential pitfall. Weights may determine advantages from its shape, not from exact score, like the following picture.

![ELF](https://user-images.githubusercontent.com/28377280/63804233-b3590380-c951-11e9-92bc-bb5ef95a5ff7.png)
<p align="center">This is from <i>ELF OpenGo v1</i>.<br>
White passed, Black to play. Black should be winning or even at least,<br>
because it has one more stone on the corner of board.</p>

**Comment by l1t1** [comment]
@bubblesld @bjiyxo could you upload a weight of yours that biggger than 40x256 for testing?

**Comment by bubblesld** [comment]
We had one 80b about one month ago. But it was only 55% against the best 40b at that time (so probably weaker than the current best) for the same 1600 visits. We think that it has to be more than 60% with the same visits to be considered as a successful one. I can upload it if we have such one later.
